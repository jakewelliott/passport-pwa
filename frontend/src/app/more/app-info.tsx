import { PassportHeader } from '../../components/common/passport-header';
import { ContactUs } from '../../components/tabs/more/contact-us';
import { DownloadParkMaps } from '../../components/tabs/more/download-park-maps';
import { Reservations } from '../../components/tabs/more/reservations';

export const AppInfo = () => {
  return (
    <div className='max-w-[500px] mx-auto'>
      <div className='my-3.5'>
        <PassportHeader />
      </div>
      <ContactUs />
      <DownloadParkMaps />
      <Reservations />
    </div>
  );
};
