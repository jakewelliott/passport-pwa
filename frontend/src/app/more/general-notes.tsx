import { NotesMiniTab } from '../locations/components/notes-minitab';

export const EditGeneralNotes = () => {
	return (
		<div className='container mx-auto'>
			<div className='flex h-[calc(100vh-1.5rem-100px)] flex-col'>
				<NotesMiniTab parkId={0} />
			</div>
		</div>
	);
};
