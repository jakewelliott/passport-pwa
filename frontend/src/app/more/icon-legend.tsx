import { LegendSections } from '@/app/more/components/legend-sections';
import { dbg } from '@/lib/debug';

export const IconLegend = () => {
    dbg('RENDER', '/more/icon-legend');
    return (
        <div className='flex flex-col gap-4'>
            <h2 className='text-center'>Legend</h2>
            <LegendSections />
            <p className='min-w-[200px]'>
                <span className='text-icon-amenities'>Kids TRACK Trail: </span>Kids in Parks is a group that creates
                outdoor adventures called TRACK trails for families all over the country. Each one has fun things to
                learn about nature as you walk. You can even earn prizes by keeping track of your adventures! Check out
                kidsinparks.com to find out more.
            </p>
        </div>
    );
};
