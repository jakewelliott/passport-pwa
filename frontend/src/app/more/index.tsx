import RoundedButton from "../../components/common/rounded-button";

export default function More() {
	return (
		<>
			<h1>More</h1>
			<RoundedButton title="TEST" color="system_black" />
		</>
	);
}
