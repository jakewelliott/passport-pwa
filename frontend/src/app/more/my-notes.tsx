import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import ListRow from '../../components/common/list-row';
import { a11yOnClick } from '@/lib/a11y';

export const MyNotes = () => {
  const [generalNotes, setGeneralNotes] = useState('');
  const navigate = useNavigate();
  const parks = [
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: '',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: '',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: 'heyyo',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes:
        ' this is some super lajnfiawlrbjhvkjg hcfjtgfkgvkjhvkyufvvvvvvvv vvvvvvvvvvvvvvvvvvvvvv vvvvvvvvvvvvvvvvvvvvvvvvvvvvvv h hvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhv hvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvh vhvhvhvhvhvhvhvhvhvhvhvhvhvhv hvhvhvhvhvhvhvhvhvhvhvhhvhv',
    },
  ];
  const parksWithNotes = parks.filter((park) => park.parkNotes && park.parkNotes.trim() !== '');

  useEffect(() => {
    const savedNotes = localStorage.getItem('generalNotes');
    if (savedNotes) {
      setGeneralNotes(savedNotes);
    }
  }, []);

  return (
    <div className='container mx-auto px-4 py-4'>
      <div className='space-y-4'>
        <div {...a11yOnClick(() => navigate('/more/my-notes/general-notes'))} className='cursor-pointer'>
          <ListRow>
            <div className='flex flex-col gap-1'>
              <h3>General Notes</h3>
              <p className='mb-2 text-gray-500 text-sm'>Last updated: Not available</p>
              <p
                className='overflow-wrap-anywhere line-clamp-3 max-w-full hyphens-auto break-words'
                style={{ hyphens: 'auto' }}
              >
                {generalNotes || 'No general notes yet'}
              </p>
            </div>
          </ListRow>
        </div>

        {/* Park Notes section */}
        {parksWithNotes.length === 0 ? (
          <p className='text-gray-600'>No park notes found.</p>
        ) : (
          parksWithNotes.map((park, index) => (
            <div
              key={index}
              {...a11yOnClick(() => navigate(`/locations/location-detail/${park.abbreviation}?tab=notes`))}
              className='cursor-pointer'
            >
              <ListRow>
                <div className='flex flex-col gap-1'>
                  <h3>{park.name}</h3>
                  <p>{park.address[0].city}</p>
                  <p className='mb-2 text-gray-500 text-sm'>Last updated: Not available</p>
                  <p
                    className='overflow-wrap-anywhere line-clamp-3 max-w-full hyphens-auto break-words'
                    style={{ hyphens: 'auto' }}
                  >
                    {park.parkNotes}
                  </p>
                </div>
              </ListRow>
            </div>
          ))
        )}
      </div>
    </div>
  );
};
