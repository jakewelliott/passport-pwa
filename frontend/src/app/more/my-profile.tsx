import { LoadingPlaceholder } from '@/components/loading-placeholder';
import RoundedButton from '@/components/rounded-button';
import { useUser } from '@/hooks/queries/useUser';
import { dbg } from '@/lib/debug';
import type { UserProfile } from '@/types';
import { useNavigate } from 'react-router-dom';

const AccountInfo = ({ user }: { user: UserProfile }) => {
    return (
        <div>
            <p>Username: {user.username}</p>
            <p>Role: {user.role}</p>
            {/* <p>Account created on: {DateHelper.toStringLong(user.createdAt)}</p> */}
        </div>
    );
};

export default function MyProfileScreen() {
    dbg('RENDER', '/more/my-profile');

    const { data: user, isLoading } = useUser();
    const navigate = useNavigate();

    const handleLogout = () => {
        const popup = window.confirm('Are you sure you want to logout?');
        if (popup) {
            navigate('/logout');
        }
    };

    const handlePasswordChange = () => {
        // TODO: implement password change
    };

    if (isLoading) return <LoadingPlaceholder what='your profile' />;

    return (
        <div className='flex flex-col gap-4'>
            <h2>My Profile</h2>
            {user && <AccountInfo user={user} />}
            <RoundedButton onClick={handlePasswordChange} title='Change Password' />
            <RoundedButton onClick={handleLogout} title='Logout' />
        </div>
    );
}
