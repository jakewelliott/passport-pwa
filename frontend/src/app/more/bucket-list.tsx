import { BucketListItem } from '@/components/tabs/more/bucket-list-item';

export const BucketList = () => {
  return (
    <div className='p-7'>
      <div className='max-w-96 mx-auto m-7'>
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
      </div>
    </div >
  );
};
