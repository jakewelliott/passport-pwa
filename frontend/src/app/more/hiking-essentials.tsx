import { HikingEssentialItem } from '@/components/tabs/more/hiking-essential-item';

export const HikingEssentials = () => {
  return (
    <div className='mx-auto max-w-4xl p-6' lang='en'>
      <h3 className='mb-5 bg-supporting_inactiveblue p-3 text-center text-system_white uppercase'>
        WHAT TO PACK FOR A HIKE
      </h3>

      <div className='flex flex-wrap justify-center gap-6'>
        <HikingEssentialItem
          imageSrc='/icons/hiking/boot.svg'
          altText='Hiking boot'
          title='HIKING BOOTS'
          description='Make sure your boots are broken in before hitting the trails to prevent blisters and sore feet!'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/water-bottle.svg'
          altText='Water bottle'
          title='WATER'
          description="You need to drink about 1 liter of water per every 4-5 miles hiked. Don't forget water for your pet!"
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/map-and-compass.svg'
          altText='Map and compass'
          title='MAP & COMPASS'
          description="Pick up a park map from the park's visitor center or download a digital version at ncparks.gov"
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/snacks.svg'
          altText='Snacks'
          title='SNACKS'
          description='You can make your own trail mix with dark chocolate, almonds, raisins and unsalted popcorn!'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/rain-gear.svg'
          altText='Rain gear'
          title='RAIN GEAR'
          description='When hiking in wet weather, avoid wearing cotton - it traps water and takes a long time to dry.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/flashlight.svg'
          altText='Flashlight'
          title='FLASHLIGHT'
          description='Due to all the trees, the trails will get darker before the rest of the park.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/first-aid-kit.svg'
          altText='First aid kit'
          title='FIRST AID KIT'
          description='Make sure your kit has bandages, antiseptic, antihistamines, and tweezers.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/matches.svg'
          altText='Matches'
          title='MATCHES'
          description='Pack waterproof matches or carry them in a waterproof container.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/knife.svg'
          altText='Knife'
          title='KNIFE'
          description='Make sure your knife or multitool has a locking blade to prevent any injuries.'
        />
      </div>
    </div>
  );
};
