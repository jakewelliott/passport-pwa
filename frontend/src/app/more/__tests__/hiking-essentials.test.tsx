import { HikingEssentials } from '@/app/more/hiking-essentials';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

describe('HikingEssentials Component - User Stories', () => {
    it('matches snapshot', () => {
        const { container } = render(<HikingEssentials />);
        expect(container).toMatchSnapshot();
    });
});
