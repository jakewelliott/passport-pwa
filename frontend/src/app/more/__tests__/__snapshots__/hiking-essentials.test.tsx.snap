// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HikingEssentials Component - User Stories > matches snapshot 1`] = `
<div>
  <div
    class="mx-auto max-w-4xl p-6"
    lang="en"
  >
    <h3
      class="mb-5 bg-supporting_inactiveblue p-3 text-center text-system_white uppercase"
    >
      WHAT TO PACK FOR A HIKE
    </h3>
    <div
      class="flex flex-wrap justify-center gap-6"
    >
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Hiking boot"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/boot.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            HIKING BOOTS
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Make sure your boots are broken in before hitting the trails to prevent blisters and sore feet!
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Water bottle"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/water-bottle.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            WATER
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            You need to drink about 1 liter of water per every 4-5 miles hiked. Don't forget water for your pet!
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Map and compass"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/map-and-compass.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            MAP & COMPASS
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Pick up a park map from the park's visitor center or download a digital version at ncparks.gov
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Snacks"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/snacks.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            SNACKS
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            You can make your own trail mix with dark chocolate, almonds, raisins and unsalted popcorn!
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Rain gear"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/rain-gear.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            RAIN GEAR
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            When hiking in wet weather, avoid wearing cotton - it traps water and takes a long time to dry.
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Flashlight"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/flashlight.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            FLASHLIGHT
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Due to all the trees, the trails will get darker before the rest of the park.
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="First aid kit"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/first-aid-kit.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            FIRST AID KIT
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Make sure your kit has bandages, antiseptic, antihistamines, and tweezers.
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Matches"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/matches.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            MATCHES
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Pack waterproof matches or carry them in a waterproof container.
          </p>
        </div>
      </div>
      <div
        class="w-[calc(50%-1rem)] min-w-[150px] max-w-[200px] rounded-lg"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="relative aspect-[1/.5] w-full rounded-t-full bg-supporting_lightblue"
          >
            <img
              alt="Knife"
              class="absolute inset-0 h-[calc(100%+6px)] w-full object-contain"
              src="/icons/hiking/knife.svg"
            />
          </div>
          <h4
            class="min-h-[20px] w-full hyphens-auto break-words bg-supporting_inactiveblue p-1.5 text-center text-[clamp(11px,calc(24/550*100vw),24px)] text-system_white leading-[clamp(8px,calc(20/550*100vw),20px)]"
          >
            KNIFE
          </h4>
          <p
            class="w-full bg-supporting_lightblue p-mini py-3 text-center text-[clamp(12px,calc(18/550*100vw),18px)] leading-[clamp(16px,calc(22/550*100vw),22px)]"
          >
            Make sure your knife or multitool has a locking blade to prevent any injuries.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
